vignettes/articles/scripting-tasks-config.Rmd

```diff
@@ -0,0 +1,387 @@
+  is_step_ahead = TRUE,
+  time_unit = "week"
+)
+
+target_metadata_death <- create_target_metadata_item(
+  target = "inc death",
+  target_name = "Weekly incident influenza deaths",
+  target_units = "rate per 100,000 population",
+  target_keys = list(target = "inc death"),
+  target_type = "discrete",
+  is_step_ahead = TRUE,
+  time_unit = "week"
+)
+
+target_metadata <- create_target_metadata(target_metadata_hosp, target_metadata_death)
+```
+
+If you inspect the `target_metadata_hosp` object, you will see that it appears
+as a class `target_metadata_item` with additional attributes about the schema
+that created it:
+
+```{r tmh}
+str(target_metadata_hosp)
+```
+
+
+Likewise, `target_metadata` is a combination of `target_metadata_hosp`,
+and `target_metadata_death`:
+
+
+```{r, tme}
+str(target_metadata)
+```
+
+
+If this process just creates lists with metadata, then why bother using `hubAdmin` functions at all to create them?
+The benefit is that `hubAdmin` `create_*()` functions provide some basic validation of objects when creating them, helping you catch some potential mistakes sooner. For
+example, some of the values are interdependent and if you accidentally leave one
+out, the function will provide a helpful error:
+
+```{r tmh-example}
+#| error: true
+create_target_metadata_item(
+  target = "inc hosp",
+  target_name = "Weekly incident influenza hospitalizations",
+  target_units = "rate per 100,000 population",
+  target_keys = list(target = "inc hosp"),
+  target_type = "discrete",
+  is_step_ahead = TRUE
+)
+```
+
+Now that we have defined the targets in `target_metadata`, we can move on
+to defining the task ID objects, which will define what the modeling parameters
+will be.
+
+### Creating the `task_id` objects
+
+In this modeling effort, we want to **require** modelers to provide week-ahead
+predictions for the "weekly incident influenza hospitalizations" target
+(`inc hosp`). Model submissions can optionally include five states, provide up
+to 4-week-ahead predictions, and also provide predictions for the "weekly
+incident influenza deaths" target (`inc death`).
+
+```{r set-task-ids}
+origin_date <- create_task_id(
+  "origin_date",
+  required = NULL,
+  optional = c("2023-01-02", "2023-01-09", "2023-01-16")
+)
+
+location <- create_task_id(
+  "location",
+  required = "US",
+  optional = c("01", "02", "04", "05", "06")
+)
+
+horizon <- create_task_id("horizon", required = 1L, optional = 2:4)
+
+target <- create_task_id("target", required = "inc hosp", optional = "inc death")
+
+task_ids_example <- create_task_ids(origin_date, location, horizon, target)
+
+```
+
+Now that we've created the task IDs that specify the modeling tasks, we can define the outputs we
+expect from the models. 
+
+### Creating the `output_type` objects
+
+For this example, we want to have three output type objects:
+
+1. a required `"mean"` output type with a "double" value
+2. a required `"quantile"` output type with a "double" value
```

Comment:
Very minor point -- but it might be good to use an integer data type for the quantile predictions in this example.
```suggestion
2. a required `"quantile"` output type with a "quantile" value
```

Reply:
See my first comment about my own problems with this narrative. As I mentioned, I think this is a "good enough" first start to get things moving. I will open an issue to fix the narrative problems later.

Reply:
so I'm going to resolve this and we can demonstrate that quantiles can also be integers later when we solve the other narrative problems.